test/inductor/test_torchinductor.py

```diff
@@ -4200,6 +4201,89 @@ def forward(pred_objectness_logits_3_: torch.Tensor):
             result = forward(*args)
             assert same(result, torch.sort(args[0], descending=True, dim=1)[0])
 
+    class TritonCodeGenTests(TestCase):
```

Comment:
is this really the best way to test this? My understanding is that this annotation is used for optimization somewhere in Triton and that a correctness test will not necessarily verify that the config has been updated as expected. As a result I thought looking at the code that is generated would be the best way to verify the expected behavior. I didn't see a code text comparison suite so I added a test suite that extracts the kernels given a function so that I could peek at the configs

Reply:
That is correct.  Wrong annotations *might* work, so a more targeted unit test is needed.  Overall this way of testing it seems fine to me, though I think it could be done in less code.